PySDM/dynamics/relaxed_velocity.py

```diff
@@ -33,21 +64,28 @@ def register(self, builder):
         )
         self.terminal_vel_attr: Attribute = builder.get_attribute("terminal velocity")
         self.volume_attr: Attribute = builder.get_attribute("volume")
+        self.radius_attr: Attribute = builder.get_attribute("radius")
 
         self.rho_w: float = builder.formulae.constants.rho_w  # TODO #798
 
-        self.tmp_data = self.particulator.Storage.empty(
-            (self.particulator.n_sd,), dtype=float
-        )
+        self.tmp_momentum_diff = self.create_storage(self.particulator.n_sd)
+        self.tmp_tau = self.create_storage(self.particulator.n_sd)
+        self.tmp_scale = self.create_storage(self.particulator.n_sd)
 
     def __call__(self):
-        self.tmp_data.product(self.terminal_vel_attr.get(), self.volume_attr.get())
-        self.tmp_data *= (
-            self.rho_w  # TODO #798
+        # calculate momentum difference
+        self.tmp_momentum_diff.product(
+            self.terminal_vel_attr.get(), self.volume_attr.get()
+        )
+        self.tmp_momentum_diff *= (
+            self.rho_w
         )  # TODO #798 - we plan to use masses instead of volumes soon
-        self.tmp_data -= self.fall_momentum_attr.get()
-        self.tmp_data *= 1 - exp(-self.particulator.dt / self.tau)
+        self.tmp_momentum_diff -= self.fall_momentum_attr.get()
+
+        self.calculate_tau(self.tmp_tau, self.radius_attr.get())
```

Comment:
this is only needed if `not self. constant`, right?

Reply:
yep! I fixed this so it only calls `calculate_tau` once if `self.constant`.